### PR TITLE
Add configuration porter

### PR DESCRIPTION
**Pull request by fsimonis**
## Main changes of this PR

This PR adds a simple configuration porter that applies a range of transformations on the xml file and outputs the result.

The script backups the old configuration and overwrites the old one.

## Motivation and additional information

Reduce unnecessary work for users.

## Author's checklist

* [ ] I used the [`pre-commit` hook](https://precice.org/dev-docs-dev-tooling.html#setting-up-pre-commit) to prevent dirty commits and used `pre-commit run --all` to format old commits.
* [ ] I added a changelog file with `make changelog` if there are user-observable changes since the last release.
* [ ] I added a test to cover the proposed changes in our test suite.
* [ ] For breaking changes: I documented the changes in the appropriate [porting guide](https://precice.org/couple-your-code-porting-overview.html).
* [ ] I sticked to C++17 features.
* [ ] I sticked to CMake version 3.16.3.
* [ ] I squashed / am about to squash all commits that should be seen as one.

## Reviewers' checklist

* [ ] Does the changelog entry make sense? Is it formatted correctly?
* [ ] Do you understand the code changes?

### CODE REVIEW

File: tools/porting/precice-2-to-3 [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    for m in filter(lambda c: c.tag.startswith("mapping"), p.iterchildren()):
+        # scaled-consisitent constraint
+        if m.attrib["constraint"] == "scaled-consisitent":
+            m.attrib["constraint"] = "scaled-consisitent-surface"
+
+        # rbf mappings
+        if m.tag.startswith("mapping:rbf"):
+            rbf = m.tag.removeprefix("mapping:rbf-")
+            if isTrue(attrib.get("use-qr-decomposition", "false")):
+                m.tag = "mapping:rbf-global-iterative"
```
Comment (davidscn) [inline tools/porting/precice-2-to-3:52]: I would just go here for `mapping:rbf` (in both branches). That's what we want the users to configure.

File: tools/porting/README.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# Porting Tools
+
+## Porting configuration from version 2 to version 3
+
+The tool `precice-2-to-3` ports a configuration file of precice version 2 to version 3.
+It requires python 3.6 and the python package `lxml` to be installed on the system.
+
+You may still need to adjust the resulting configuration file, thus it is highly recommended to read the porting guide.
```
Comment (uekerman) [inline tools/porting/README.md:8]:
```suggestion
You may still need to adjust the resulting configuration file, thus it is highly recommended to read [the porting guide](https://precice.org/couple-your-code-porting-v2-3.html).
```

**Comment by fsimonis** [comment]
Turns out that lxml doesn't allow to generate tags with prefixes. One needs to define a namespace.

So this is getting cumbersome

**Comment by fsimonis** [comment]
I'll close this. Handling our namespaces is very messy.

**Pull request closed without merging**